src/bootstrap/src/utils/exec.rs

```diff
@@ -79,6 +98,8 @@ impl<'a> BootstrapCommand {
     }
 
     pub fn arg<S: AsRef<OsStr>>(&mut self, arg: S) -> &mut Self {
+        let arg_str = arg.as_ref().to_string_lossy().into_owned();
+        self.args.push(arg_str.clone());
         self.command.arg(arg.as_ref());
```

Comment:
Because of `as_command_mut`, someone could take the inner Command and set properties on it that won't be included in the cache key. Until we can get rid of that function, we should probably make the cache optional, in the sense that we remember that we shouldn't cache if:
1) `as_command_mut` is ever called (or, as an alternative, we can change it to `into_command_mut(self)`, but that would be a quite massive change across bootstrap).
2) When `impl From<Command> for BootstrapCommand` is used (I checked and there is at least one place where we can't get rid of this impl, so we need to turn off caching in that case).